packages/ui-kit/src/components/mermaid-lazy.tsx

```diff
@@ -0,0 +1,16 @@
+import { lazy } from 'react';
+
+const mermaidVersion = '9.3';
+
+const Mermaid = lazy(() => {
```

Comment:
Shouldn't we simply load the `mermaid-client-side` module here?

```tsx
const Mermaid = lazy(() => import('./mermaid-client-side'));
```

Then in the `mermaid-client-side` component you use the `useScript` hook.

Reply:
Example usage: https://javascript.plainenglish.io/how-to-load-a-dynamic-script-in-react-2940d30998dd

Reply:
We could replicate the same loading indicator in the component through checking the loading state vs. through suspense.  

I had issues getting the tests to properly wait until the script is loaded and the component rendered when doing it through useScript and condiditional rendering, so I tried this other approach that works through suspense.  But it may well be that it can be solved in the test code, too.

Reply:
I reverted to your initial implementation.